### PR TITLE
sync: Timeline semaphore support for syncval

### PR DESCRIPTION
**Pull request by artem-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7600

### CODE REVIEW

**Comment by artem-lunarg** [comment]
Removed malformed tests: violated rule increasing signal values. Caught by AMD driver which reported error from QueueSubmit.

File: tests/unit/sync_val.cpp [hunk 1]
```diff
@@ -5709,63 +5709,6 @@ TEST_F(NegativeSyncVal, QSDebugRegion_Secondary) {
     m_default_queue->Wait();
 }
 
-// TODO: this test should be removed after timeline semaphore support is added to sync validation
```
Comment (spencer-lunarg) [inline tests/unit/sync_val.cpp:5712]: good thing for `todo` :laughing:

**Review by spencer-lunarg — APPROVED** [review]

File: tests/unit/sync_val.cpp (hunk 1 above)
Comment (artem-lunarg) [reply tests/unit/sync_val.cpp:5712]: as was designed it stopped working, that’s how I found it

**Comment by artem-lunarg** [comment]
Found issue (unresolved batches are not getting removed) in the latest version of Granite renderer. Investigating.

**Comment by artem-lunarg** [comment]
The issue is fixed.

**Comment by artem-lunarg** [comment]
Fixed leak when resolved wait-before-signal did not retire older signals

**Comment by artem-lunarg** [comment]
Cleanup and refactoring.

**Comment by artem-lunarg** [comment]
Found one more potential issue when testing Granite renderer (unresolved batches are not getting resolved). Investigating.

**Comment by artem-lunarg** [comment]
The issue is fixed (external semaphore case).

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM. It would be good to have:
-@jzulauf-lunarg eyes on it
-A follow-on change to [docs/synchronization.md](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/docs/synchronization.md) at some point to describe how this works.

**Comment by artem-lunarg** [comment]
Additional renderers/games used for testing:
* Granite renderer (gltf-viewer-simple, latency-test)
* Tiny Glade Demo 
* Return to Monkey Island (only debug vvl build)
* CS2/Dota2 (does not use timelines but stability/memory usage checks)
* Quake 3 Arena Kenny Edition (does not use timelines but stability/memory usage checks)

The feature documentation will be added later.

**Pull request merged**